src/eko/io/types.py

```diff
@@ -0,0 +1,174 @@
+"""Common type definitions, only used for static analysis.
```

Comment:
VS code is rightly complaining about you shadowing python [`types`](https://docs.python.org/3/library/types.html?highlight=types#module-types)

Reply:
True, but it is a reflection module to mess up with Python classes creation (not type hints, since that is `typing` module... I know, it is rather confusing).

Do we really care?

Reply:
In the end, we can always import scoped, like:
```py
import eko.io.types
```
or once lifted:
```py
import eko.types
```

It is not the name of a top-level package, so it should be fine. Don't you think so?

Reply:
I'd say we can also rename and so appease VS code (i.e. have only relevant curly lines around) ... `common_types`? `ctypes`? `input_types`?

Reply:
[`ctypes`](https://docs.python.org/3/library/ctypes.html?highlight=ctypes#module-ctypes)